src/content/guides/development.md

```diff
@@ -265,6 +266,34 @@ __webpack.config.js__
 
 This tells `webpack-dev-server` to serve the files from the `dist` directory on `localhost:8080`.
 
+`webpack-dev-server` uses `output.path` as the root path for serving generated files. Generated files will be available under `http://localhost:[devServer.port]/[output.publicPath]/[output.filename]`. If you want to serve generated files under a specific directory on dev server, e.g. under the `assets` directory (`http://localhost:8080/assets/bundle.js`), you will need to set output directory with `output.filename` instead of `output.path`, or configure `output.publicPath`.
+
+__webpack.config.js__
+
+``` diff
+ const path = require('path');
+
+ module.exports = {
+   // Set output directory with output.filename instead of output.path
+   output: {
+-    filename: 'bundle.js',
++    filename: 'assets/bundle.js',
+-    path: path.resolve('assets'),
++    path: path.resolve(),
+   },
+ };
+
+ // Or
+
+ module.exports = {
+   output: {
+     filename: 'bundle.js',
+     path: path.resolve('assets'), // Keep this line intact to specify output directory for building
++    publicPath: '/assets/', // Tell webpack-dev-server to serve generated files under assets directory
+   },
+ };
+```
+
```

Comment:
How about the alternative way to setup dev-server, with `output.path` and `output.filename`?
``` diff
 module.exports = {
   output: {
     filename: 'bundle.js',
     path: path.resolve('assets'), // Keep this line intact to specify output directory for building
+    publicPath: '/assets/', // Tell webpack-dev-server to serve generated files under assets directory
   },
 };
```

Reply:
Sorry, I'm a little confused by your example, why would you want to set the `output.publicPath` when developing? As far as I know, you can just leave it unset and everything should work just fine.

Reply:
Ok, I just took some time to check your example. In your original example here https://github.com/webpack/webpack-dev-server/issues/2982, no `publicPath` is provided, and it defaults to `auto`, webpack dev server would transform it into `/` as per https://github.com/webpack/webpack-dev-server/blob/788f2ba7d9152610ba0d3024f0079c98aad86991/lib/Server.js#L632, hence the `app.bundle.js` is served from `/app.bundle.js` instead of `./assets/js/app.bundle.js`. I think you should consider improve your project structure first. What we're documenting here, I would say it's bad practice, we can easily shoot ourself in the foot.

Reply:
Anyway, if you insist these should be added, I'll leave it to other reviewers.

Reply:
Sorry I referenced the wrong code snippet. I have just updated it.

It doesn't feel so right as defining both `output.path` and `devServer.publicPath` seems like adding duplicated logic to the config.

It is your call though as maintainers have more knowledge to their projects.

Reply:
Generally, we don't have to worry about `output.publicPath`.

Here's what I know about `output.path` and `output.publicPath`. With tools like webpack, we actually have more than one environments, for example, where we build the project could be different from where we deploy the project.

Let's say I'm going to build a project `/users/sam/project` on my computer.

1. `output.path` decides where we build the project to, e.g., `/users/sam/project/dist`
2. now I want to deploy all files under `dist` to a production server under domain `https://example.com`, I can just copy all files under local `dist` to the root of my domain, e.g., `/users/ubuntu/site_a/`. Everything's fine so far.
3. however, I change my mind later and decide to deploy my project under `https://example.com/blog/`, with the default `output.publicPath`, I can copy files to `/users/ubuntu/site_a/blog/` directory, and everything should just work fine too.
4. Some times later, I decide to host assets like image/css/js etc. on a CDN service, we copy those assets to cdn and we can access them from address like `https://cdn/what/ever/path/main.js`. However the `index.html` is still on our own server, i.e., we want the `index.html` to load assets from cdn instead of our own server, now we need to set `output.publicPath` to `https://cdn/what/ever/path/`, and the html file webpack generated will reference `https://cdn/what/ever/path/main.js` instead of `https://example.com/blog/main.js`.

Reply:
OIC. webpack is designed for building all front end files at once. In most cases, the html files that referencing css and js files will also process by webpack and located at `output.path`.

In that case there won't be any duplicated logic in the config.

Reply:
By the way, can you help merge the pull request if it is OK?